reviewer/reviewer.go

```diff
@@ -0,0 +1,264 @@
+	}
+
+	return allUsers, nil
+}
+
+func listAllOrgMembers(ctx context.Context, client *github.Client, org string) ([]string, error) {
+	opt := &github.ListMembersOptions{
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+		},
+	}
+
+	// get all pages of results
+	var allUsers []string
+
+	for {
+		users, resp, err := client.Organizations.ListMembers(ctx, org, opt)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to list members of org %s page %d", org, opt.Page)
+		}
+		for _, u := range users {
+			allUsers = append(allUsers, u.GetLogin())
+		}
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allUsers, nil
+}
+
+func listAllCollaborators(ctx context.Context, client *github.Client, org, repo string) ([]string, error) {
+	opt := &github.ListCollaboratorsOptions{
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+		},
+	}
+
+	// get all pages of results
+	var allUsers []string
+
+	for {
+		users, resp, err := client.Repositories.ListCollaborators(ctx, org, repo, opt)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to list members of org %s page %d", org, opt.Page)
+		}
+		for _, u := range users {
+			allUsers = append(allUsers, u.GetLogin())
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allUsers, nil
+}
+
+// select n random values from the list of users
+func selectRandomUsers(n int, users []string) []string {
+	generated := map[int]struct{}{}
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	var selections []string
+	if n == 0 {
+		return selections
+	}
+	if n >= len(users) {
+		return users
+	}
+
+	for i := 0; i < n; i++ {
+		for {
+			i := r.Intn(len(users))
+			if _, ok := generated[i]; !ok {
+				generated[i] = struct{}{}
+				selections = append(selections, users[i])
+				break
+			}
+		}
+	}
+	return selections
+}
+
+func shoveIntoMap(m map[string]struct{}, u []string) map[string]struct{} {
+	for _, n := range u {
+		m[n] = struct{}{}
+	}
+	return m
+}
+
+func FindRandomRequesters(ctx context.Context, prctx pull.Context, result common.Result, client *github.Client) ([]string, error) {
+	logger := zerolog.Ctx(ctx)
+	pendingLeafNodes := findLeafChildren(result)
+	var requestedUsers []string
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	logger.Debug().Msgf("Collecting reviewers for %d pending leaf nodes", len(pendingLeafNodes))
+
+	for _, child := range pendingLeafNodes {
+		allUsers := make(map[string]struct{})
+		allUsers = shoveIntoMap(allUsers, child.RequestedUsers)
+
+		if len(child.RequestedTeams) > 0 {
+			randomTeam := child.RequestedTeams[r.Intn(len(child.RequestedTeams))]
+			teamInfo := strings.Split(randomTeam, "/")
+			team, _, err := client.Teams.GetTeamBySlug(ctx, teamInfo[0], teamInfo[1])
+			if err != nil {
+				logger.Debug().Err(err).Msgf("Unable to get member listing for team %s", randomTeam)
+				//return nil, errors.Wrapf(err, "Unable to get information for team %s", randomTeam)
```

Comment:
Are these commented out so the process is best-effort? If so, it would be good to avoid some of this nesting by putting each sub-selection type (team, organization) into a helper function that can return on error.

Reply:
Yep, exactly. I felt like this might friendly, since policy-bot will fail here if the app is not installed. You're right that I should move this into a helper.